web/frontend/valexa_app/src/components/ProfileTable.vue

```diff
@@ -0,0 +1,131 @@
+<template>
+<table>
+	<tbody>
+		<tr>
+			<th>File name</th>
+			<th colspan="7" style="text-align: center;">static Bifenthrin Level 01 to 5.xlsx static</th>
+		</tr>
+		<tr>
+			<td>Modèle</td>
+			<td colspan="3">{{profile.model.name}}</td>
+			<td>beta</td>
+			<td>{{ profile.tolerance_limit }}</td>
+			<td>t</td>
+			<td>{{ profile.acceptance_limit }}</td>
+		</tr>
+		<tr>
+			<td>LOQ Min</td>
+			<td colspan="3">{{ profile.min_lq }}</td>
+			<td>LOQ Max</td>
+			<td colspan="3">{{ profile.max_lq }}</td>
+		</tr>
+		<tr>
+			<td>LOD</td>
+			<td colspan="3">{{ profile.ld }}</td>
+			<td>Corr. Factor.</td>
+			<td>{{ profile.model.correction_factor }}</td>
+			<td>Recovery</td>
+			<td>static 1428.5714285714284 static</td>
+		</tr>
+		<tr>
+			<td colspan="8"></td>
+		</tr>
+		<tr>
+			<td rowspan="2">Concentration</td>
+			<td rowspan="2">Mesuré</td>
+			<td colspan="2">Exactitude</td>
+			<td colspan="2">Précision</td>
+			<td colspan="2">Justesse</td>
+		</tr>
+		<tr>
+			<td>Biais absolue</td>
+			<td>Biais relatif</td>
+			<td>Répétabilité (%%RSD)</td>
+			<td>Intermédiaire (%%RSD)</td>
+			<td>Limite de tolérance</td>
+			<td>Limite relative</td>
+		</tr>
+		<tr v-for="(level, i) in profile.levels" v-bind:key="i"> 
+			<td>{{round(level["introduced_concentration"], 3)}}</td>
+			<td>{{round(level["calculated_concentration"], 3)}}</td>
+			<td>{{round(level["bias"], 3)}}</td>
+			<td>{{round(level["relative_bias"], 2)}}</td>
+			<td>{{round(level["repeatability_var"], 3)}}</td>
+			<td>{{round(level["repeatability_std"], 3)}}</td>
+			<td>{{round(level["abs_tolerance"], 2)}}</td>
+			<td>{{round(level["rel_tolerance"], 2)}}</td>
+		</tr>
+		<tr>
+			<td colspan="8"></td>
+		</tr>
+	</tbody>
+</table>
+</template>
+
+<script>
+
+export default {
+	name: "ProfileTable",
+	//TODO stricter type control 
+  props: {
+    profile: Object,
+  },
+  data() {
+    return {
+
+    };
+	},
+	methods: {
+		round(value, decimals) {
+			if (typeof value === "object") {
+				let newValue = [];
+				for (let v of value) {
+					newValue.push(+Number(Math.round(v +'e'+ decimals) +'e-'+ decimals).toFixed(decimals));
+				}
+				return newValue;
+			} else {
+				return Number(Math.round(value +'e'+ decimals) +'e-'+ decimals).toFixed(decimals);
```

Comment:
Same here not clear. However, we duplicate the same code two times. Should be extracted with a well-named function.

Reply:
It rounds `value` to a fixed number of decimals.

Reply:
This is the `'e' + decimals` that confuses me.

Reply:
toFixed() rounds down for decimals ending with 5, but we want to round up. Math.round() rounds up to the nearest integer. Here my_round(1.005, 2) would give -`'e'+decimals`> 100.5 -`Math.round`> 101 -`'e-'+decimals`> 1.01.

But after testing I've found that it doesn't work with numbers already in scientific notation, or automatically converted to it (like `0.0000001` -> `1e-7`), so I'm going to try something else. Seriously, rounding up decimals reliably is proving to be harder than I thought

Reply:
Especially with javascript. However, did you look for example on stackoverflow?

Reply:
Yes, there is a lot of them that don't round up or can't handle scientific notation. I found two that met these criteria and put one of them in a math_utils.js file.